Add follow-up to data licensing post (#360)
epiverse-trace/epiverse-trace.github.io: posts/data-licensing-cran.qmd

---
title: "Licensing R packages with code and data: learnings from submitting to CRAN"
date: 2025-01-20
author:
  - name: "Joshua Lambert"
    orcid: "0000-0001-5218-3046"
  - name: "Chris Hartgerink"
    orcid: "0000-0003-1050-6809"
categories: [R, open-source, package development, DOI]
format:
  html: 
    toc: true
# doi: "10.59350/z78kb-qrz59"
---

::: {.callout-note}
This is a follow-up blog post to the [_Dual licensing R packages with code and data_](data-licensing.html) post published in September 2024. It contains learnings from submitting **epiparameter** to CRAN with a dual license.
:::

### Overview of previous blog post on R package licensing

> We previously published a [post on the Epiverse-TRACE blog discussing the importance of licensing code and data for open source](data-licensing.html). It covered the differences between open source licenses for code and those that are required for other forms of information, in our case, data. R packages most commonly distribute code, and therefore require an [appropriate open source license](https://cran.r-project.org/doc/FAQ/R-exts.html#Licensing). Less frequently, R packages are used to bundle and share data, so-called ["data packages"](https://r-pkgs.org/data.html), and require an applicable license for reuse and redistribution.
> 
> We recommend reading the original blog post for a more in-depth explanation of each of these points.

Reviewer: ```suggestion
We previously published a [post on the Epiverse-TRACE blog discussing the importance of licensing code and data for open source](data-licensing.html). It covered the differences between open source licenses for code and those that are required for other forms of information, in our case, data. R packages most commonly distribute code, and therefore require an [appropriate open source license](https://cran.r-project.org/doc/FAQ/R-exts.html#Licensing). Less frequently, R packages are used to bundle and share data, so-called ["data packages"](https://r-pkgs.org/data.html), and require an applicable license for reuse and redistribution. We recommend reading [the original blog post](data.licensing.html) for a more in-depth explanation of each of these points.
```

After conducting an online search, dual licensing for R packages seems rare. An interesting example of dual licensing is the [**igraphdata** package](https://github.com/igraph/igraphdata/blob/main/LICENSE), which contains several licenses: One for each dataset included in the package. Similar to igraphdata, we dual licensed the [**epiparameter** package](https://github.com/epiverse-trace/epiparameter), which until versus v0.4.0 contained both code and data. We [licensed the code using the `DESCRIPTION` file](https://github.com/epiverse-trace/epiparameter/blob/v0.3.0/DESCRIPTION#L28) and used [the `LICENSE` file](https://github.com/epiverse-trace/epiparameter/blob/v0.3.0/LICENSE) to license the data under CC0.  Concretely, we included this additional text  in `LICENSE` to clarify the dual license and that we recommend citing the original source regardless:

> All data included in the epiparameter R package is licensed under CC0 (<https://creativecommons.org/publicdomain/zero/1.0/legalcode.txt>). This includes the parameter database (extdata/parameters.json) and data in the data/ folder. Please cite the individual parameter entries in the database when used.

> However, upon submission of epiparameter to CRAN, we were rejected due to the dual licensing. The CRAN package reviewer stated:

Reviewer: ```suggestion
However, upon submission of epiparameter to CRAN, the dual licensing approach was rejected. The CRAN package reviewer stated:
```
